sycl/include/sycl/detail/string.hpp

```diff
@@ -0,0 +1,45 @@
+//==----------------- string.hpp - SYCL standard header file ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+
+class string {
+  const char *str;         // used to send existing std::string to libsycl
+  char *ret_str = nullptr; // set from libsycl
+
+public:
+  string() : str(nullptr), ret_str(nullptr) {}
+  string(const char *ptr) : str(ptr) {}
+  string(std::string strn) : str(strn.c_str()) {}
+
+  bool operator==(const char *st) { return strcmp(str, st) == 0; }
+
+  std::string marshall() { return std::string(ret_str); }
```

Comment:
I don't think this overload is needed.

Reply:
Added comment what this method is intended.
Hopefully it clarifies your doubt.